### PR TITLE
MCF file utilities for statvar processor

### PR DESCRIPTION
**Pull request by ajaits**
This is part of the commits for StatVarProcessor. 

It includes the following utilities:
- mcf_file_util.py: Unilities to read/write MCF files
- mcf_diff.py: utilities to compare MCF nodes
- mcf_filter.py: utilities to remove existing Nodes from an MCF file

### CODE REVIEW

File: scripts/statvar/mcf_diff.py [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+import difflib
+import os
+import sys
+
+from absl import app
+from absl import flags
+from absl import logging
+
+_SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
```
[the remark this answers is not in the document]
Comment (ajaits) [reply scripts/statvar/mcf_diff.py:68]: Added __init__.py.

Is there any other way to import scripts from other folders without using sys.path?

File: scripts/statvar/mcf_file_util.py [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+def strip_namespace(value: str) -> str:
+    """Returns the value without the namespace prefix.
+
+  Args:
+    value: string from which the namespace prefix is to be removed.
+
+  Returns:
+    value without the namespace prefix if there was a namespace
+
+  Any sequence of letters followed by a ':' is treated as a namespace.
```
[the remark this answers is not in the document]
Comment (ajaits) [reply scripts/statvar/mcf_file_util.py:117]: Done

File: scripts/statvar/mcf_file_util.py [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+    """Return a dict of nodes from the MCF file with the key as the dcid
+
+  and a dict of property:value for each node.
+  """
+    if not filenames:
+        return {}
+    # Load files in order of input
+    files = []
+    if isinstance(filenames, str):
+        filenames = filenames.split(',')
```
[the remark this answers is not in the document]
Comment (ajaits) [reply scripts/statvar/mcf_file_util.py:307]: Some scripts use the DEFINE_list to get the list of files while some configs have a string that have comma separated names. 
This was using `file_util.file_get_matching()` earlier that accepted list or str and returned files in sorted order, but with the overwrite option files have to be processed in the input order, hence supporting both list and str.

File: scripts/statvar/mcf_file_util.py [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+{
+  '<dcid>': OrderedDict( [
+                '<property>': '<value>'
+                ...
+                ])
+}
+where
+  <dcid> is the node dcid with the namespace prefix: 'dcid:'
+  <property> is a property name
+  <value> is a string with a single value or a comma seperated list of values.
```
[the remark this answers is not in the document]
Comment (ajaits) [reply scripts/statvar/mcf_file_util.py:26]: Currently the quoted values are handled using a csv reader, so nodes can have multiple quoted strings with comma within them.

Do you need a utility that converts such values into list?

File: scripts/statvar/mcf_diff.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (keyurva) [reply scripts/statvar/mcf_diff.py:68]: Yes. You should make util into a module as well and then you should be able to use a standard import.

**Review by keyurva — APPROVED** [review]

File: scripts/statvar/mcf_diff.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajaits) [reply scripts/statvar/mcf_diff.py:68]: thanks for sharing the module tips.
it works as a module too when invoke dform data:
`python -m scripts.statvar.mcf_file_util --input_mcf=scripts/statvar/test_data/sample_output_stat_vars.mcf --output_mcf=/tmp/t.csv`

**Pull request merged**